File: Data Structures & Algorithms/longest-common-prefix/submission-0.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        count = 0
        
        while True:
            for s in strs:
                if count >= len(s) or s[count] != strs[0][count]:
                    return strs[0][:count]
            count += 1

File: Data Structures & Algorithms/counting-bits/submission-0.cpp
class Solution {
public:
    vector<int> countBits(int n) {
        vector<int> res;
        for (int target = 0; target < n + 1; target++)
        {
            int count = 0;
            int x = target;

            while (x) {
                count += x & 1;
                x >>= 1;
            }
            res.push_back(count);
        }
        return res;
    }
};
